addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -873,6 +873,14 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
         if (select) {
             select.style.display = '';
         }
+        const dependencyEl = this._getDependencyEl();
+        if (dependencyEl) {
+            const fieldName = this.$target[0].querySelector('.s_website_form_input').name;
+            if (dependencyEl.closest('.s_website_form_field').dataset.visibilityDependency === fieldName) {
+                this._deleteConditionalVisibility(this.$target[0]);
+                alert(_t("You can't have 2 field visibility depending of each others. One of the 2 conditions has been removed."));
```

Comment:
`alert()` is definitely not ideal, but seems reasonable given the low probability of anyone ever seeing it.
I'd have gladly replaced it by a `Dialog.warning` or `displayNotification` but since there is a page reload after save, it doesn't work, the message is barely noticeable.
Unless you know a better way that could await the user click on the confirmation.
I guess I could return a promise as cleanForSave is async and then use an Odoo dialog which resolved the promise when clicked on dialog confirm?